src/basic/socket-util.c

```diff
@@ -1016,39 +1017,75 @@ int send_one_fd_sa(
         struct msghdr mh = {
                 .msg_name = (struct sockaddr*) sa,
                 .msg_namelen = len,
+                .msg_iov = iov,
+                .msg_iovlen = iovlen,
                 .msg_control = &control,
                 .msg_controllen = sizeof(control),
```

Comment:
So `msg_control` and `len` are initialized here, but reset below. Please remove the initialization (to let it be zero), and not set it below if not necessary. (I see that this is copied from existing code... But it should be fixed when anyway.)

Reply:
Done. Yeah, this looks better without the else branch.

I still need to set msg_controllen twice, since the CMSG_FIRSTHDR() macro depends on it, but for the case where there's no FD it won't be set at all.

Thanks!

Reply:
This was addressed in latest version.